extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/jaxrs/ResteasyReactiveViolationExceptionMapper.java

```diff
@@ -56,17 +64,24 @@ private boolean isReturnValueViolation(ConstraintViolation<?> violation) {
         return secondNode.getKind() == ElementKind.RETURN_VALUE;
     }
 
-    protected Response buildResponse(Object entity, Status status) {
-        return Response.status(status).entity(entity).build();
-    }
-
     private Response buildViolationReportResponse(ConstraintViolationException cve) {
+        Status status = Status.BAD_REQUEST;
+        Response.ResponseBuilder builder = Response.status(status);
+        builder.header(Validation.VALIDATION_HEADER, "true");
+
+        // Check standard media types.
+        MediaType mediaType = ValidatorMediaTypeUtil.getAcceptMediaType(
+                headers.getAcceptableMediaTypes())
```

Comment:
I can't use the produces from `ResteasyReactiveRequestContext` because it seems that a `fake` text/plain media item is being added (when return type is string).